src/precompile.c

```diff
@@ -50,12 +50,13 @@ void write_srctext(ios_t *f, jl_array_t *udeps, int64_t srctextpos) {
                 ios_t *srctp = ios_file(&srctext, depstr, 1, 0, 0, 0);
                 if (!srctp) {
                     jl_printf(JL_STDERR, "WARNING: could not cache source text for \"%s\".\n",
-                            jl_string_data(dep));
+                              depstr);
                     continue;
                 }
-                size_t slen = jl_string_len(dep);
+                jl_value_t *depalias = jl_fieldref(deptuple, 5);  // file @depot alias
```

Comment:
This here is responsible for the uuid mismatch. Changing it to load from 1 (= absolute path) causes `parse_cache_header` to read the write `uuid`. The corresponding change in `staticdata.c` does not cause issues.